### ISSUE TITLE
UDM Pro v2.4.23 site-to-site

### ISSUE THREAD

**Issue by sonnylu** [tusc/wireguard-kmod#73]
I have no issue reporting yet. However, I'd like to know if the site-to-site is still supported before I upgrade my UDM Pro. As it seems to be, Unifi has a custom version of wireguard.

**Comment by savek-cc** [comment]
It might be - but the 2.4. version already includes wireguard kernel modules and userspace tools - so there is no need to install any custom modules.
Be aware that the upgrade from 1.x to 2.4.x wipes your complete custom setup - so do perfom a backup.
Afterwards, you should be able to just use your existing config from the backup (wg0.conf or whatever) and resume working.

**Comment by pwallner** [comment]
I read there is no wireguard in the kernel modules and I also didn't find it in 2.4.27. I requested GPL Source for 2.4.27 from Ubiquiti. Then it should be possible to compile it to this new version. I lost all my compiling things on UDM during automatic update to 2.4.27, so from my side I can't compile it soon, now I have to restore a lot of things...

**Comment by savek-cc** [comment]
@pwallner on my UDM Pro (though updated to latest preview network app), I could dmesg|grep wireguard and see the module (maybe static/compiled in?). Also the wg and wg-quick commands are available.

**Comment by pwallner** [comment]
@savek-cc yes the old version, but missing one kernel module. So if you want the latest version you need the missing kernel module and an update acc latest firmware version. I requested the GPL archive from them, then it's possible to compile. Maybe an existing kernel is working because they have now - imho - "one" firmware for all devices.

**Comment by savek-cc** [comment]
Ok - the updated Firmware had everything I needed to get my Site-to-Site wg back up and running; did not check about newer versions or features though.
